Rewrite Log and Observability (#1830)
emqtt/emq-docs-cn: en_US/observability/alarms.md

{% endemqxee %}

## Get Alarms

> EMQX provides you with various ways to get alarms and check the detailed alarm formation. One way is to view the alarms on EMQX Dashboard, where you can view a list of active or historical alarms. However, it is only a central place for easy access to an overview of alarms that have been triggered. Another way is to subscribe to system topics through MQTT to receive real-time notifications of alarms with detailed alarm information. Alarms can also be accessed from the log or via HTTP API. 

Reviewer: ```suggestion
EMQX provides you with various ways to get alarms and check the detailed alarm formation. One way is to view the alarms on EMQX Dashboard, where you can view a list of active or historical alarms. However, it is only a central place for easy access to an overview of alarms that have been triggered. Another way is to subscribe to system topics through MQTT to receive real-time notifications of alarms with detailed alarm information. Alarms can also be accessed from the log or via REST API.
```

Taking the alarm of high system memory usage as an example, you will receive an alarm message like below:

<img src="./assets/alarm_activate_msg.png" alt="alarm massage" style="zoom:50%;" />

One system multifunction will be repeatedly reported. That is, if one alarm on high CPU usage is activated, the system will not generate another alarm of the same type. The generated alarm will be automatically deactivated when the monitored metric returns to normal, or you can manually deactivate the alarm.

### Get Alarms from Log

The activation and deactivation of alarms can be written to log (console or file). When failures occur during message transmission or event processing, detailed information can be logged, and the logging system can also be used to capture alerts through log analysis. The following example shows the detailed alarm information printed in the log:
The log level is `warning`, and the `msg` field is `alarm_is_activated` and `alarm_is_deactivated`.
<img src="./assets/view-alarms-log.png" alt="view-alarms-log" style="zoom:50%;" />

### Get Alarms via REST API

> You can query and manage alarms through the [API Docs](https://docs.emqx.com/en/enterprise/v5.0/admin/api-docs.html). Click **Alarms** on the left navigation menu on the UI to execute this API request. For how to work with EMQX API, see [REST API](../admin/api.md).

Reviewer 2: ```suggestion
You can query and manage alarms through the API. For how to work with EMQX API, see [REST API](../admin/api.md).
```

> You can query and manage alarms through the [API Docs](https://docs.emqx.com/en/enterprise/v5.0/admin/api-docs.html). Click **Alarms** on the left navigation menu on the UI to execute this API request. For how to work with EMQX API, see [REST API](../admin/api.md).

Reviewer 2: ```suggestion
You can query and manage alarms through the API. For how to work with EMQX API, see [REST API](../admin/api.md).
```

Taking the alarm of high system memory usage as an example, you will receive an alarm message like below:

<img src="./assets/alarm_activate_msg.png" alt="alarm massage" style="zoom:50%;" />

One system multifunction will be repeatedly reported. That is, if one alarm on high CPU usage is activated, the system will not generate another alarm of the same type. The generated alarm will be automatically deactivated when the monitored metric returns to normal, or you can manually deactivate the alarm.

### Obtain Alarms from Log

The activation and deactivation of alarms can be written to log (console or file). When failures occur during message transmission or event processing, detailed information can be logged, and the logging system can also be used to capture alerts through log analysis. The following example shows the detailed alarm information printed in the log:

> 

Reviewer: ```suggestion
The log level is `warning`, and the `msg` field is `alarm_is_activated` and `alarm_is_deactivated`.
```

<img src="./assets/view-alarms-api.png" alt="view-alarms-api" style="zoom:45%;" />

## Alarm Configuration

Alarm configuration includes configuring alarm settings and thresholds. Alarm settings determine how the alarm message is displayed and stored, while alarm thresholds establish limits or values that trigger the alarm when potential problems are detected. The alarm configuration feature allows you to customize the alarm settings and thresholds to meet your business needs.

### Configure Alarm Settings

The settings for alarms can only be configured by modifying the configuration items in `emqx.conf` file. The following table lists the configuration items available for alarm setting configuration.

| Configuration Item    | Description                                                  | Default Value        | Optional Values |
| --------------------- | ------------------------------------------------------------ | -------------------- | --------------- |

> | alarm.actions         | Actions of writing the alarm to log (console or file) and publishing the alarm as an MQTT message to the system topics `$SYS/brokers/emqx@xx.xx.xx.x/alarms/activate` and `$SYS/brokers/emqx@xx.xx.xx.x/alarms/deactivate`. The actions are triggered when the alarm is activated or deactivated. | `["log", "publish"]` | -               |

Reviewer: ```suggestion
| alarm.actions         | Actions of writing the alarm to log (console or file) and publishing the alarm as an MQTT message to the system topics `$SYS/brokers/<node_name>/alarms/activate` and `$SYS/brokers/<node_name>/alarms/deactivate`. The actions are triggered when the alarm is activated or deactivated. | `["log", "publish"]` | -               |
```
